server.go

```diff
@@ -552,6 +554,27 @@ func NumStreamWorkers(numServerWorkers uint32) ServerOption {
 	})
 }
 
+// RecvBufferPool returns a ServerOption that configures the server
+// to use the provided shared buffer pool for parsing incoming messages. Depending
+// on the application's workload, this could result in reduced memory allocation.
+//
+// If you are unsure about how to implement a memory pool but want to utilize one,
+// begin with grpc.NewsimpleSharedBufferPool.
+//
+// Note: The shared buffer pool feature will not be active if any of the following
+// options are used: StatsHandler, EnableTracing, or binary logging. In such
```

Comment:
I wonder why this optimization is not going to work with `StatsHandler`? I'm not very familiar with the codebase, but I couldn't see anything that'd explain this in the MR. I'm using `StatsHandler` for metrics and for some other things, but I also want my app to benefit from this optimization because currently most memory allocations in my app come from protobuf (https://github.com/golang/protobuf/issues/1495) and gRPC.

Reply:
I don't think there's any way around this, given our current API, unfortunately.

Reply:
@hueypark Opened https://github.com/grpc/grpc-go/issues/6660 to track this as a feature request. Thank you.